src/file_system/error.rs

```diff
@@ -0,0 +1,35 @@
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Error {
+    Label(Label),
+    Path(Path),
+}
+
+impl From<Label> for Error {
+    fn from(err: Label) -> Self {
+        Error::Label(err)
+    }
+}
+
+impl From<Path> for Error {
+    fn from(err: Path) -> Self {
+        Error::Path(err)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Path {
+    Empty,
+    Label(Label),
```

Comment:
I understand that this enum is for future extensibility but it feels funky to have a `Label` variant in both `Error` and `Path` enums. It will lead to more complexity on the caller side.

Reply:
Good point :ok_hand: It did feel a bit janky writing it. Cleaned up now :+1: